Apply best practices

nikola-rados opened the pull request:
Changes:
- Update names of `workflows`/`steps`
- Adjust triggers
- Remove `.travis.yml` and `Jenkinsfile`
- Update `README.md`
- Tagged tests lasting longer than 1 second as "slow"
- Removed `--cov` check from "fast" suite

Notes:
1. The `Pypi Publishing` badge will not properly appear on `README.md` since that workflow has not run.  To give it a target and to help ensure it is working correctly I was hoping to run through the release process once this has been merged in.
2. Old results using outdated names in the actions tab cannot be removed just yet.  From what I've read it is coming down the pipeline though.  That will clean up our Actions page a bit once it is implemented.
3. I removed coverage from the "fast" suite because skipping the tests brought us below the threshold.

Here is a quick look at the time we are saving by ignoring these tests:
```bash
(venv) nrados@pcic-2024:~/github/pcic/crmprtd$ pytest
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.9, pytest-3.6.3, py-1.6.0, pluggy-0.6.0
rootdir: /home/nrados/github/pcic/crmprtd, inifile: pytest.ini
plugins: requests-mock-1.7.0, mock-1.11.1, flake8-1.0.1, cov-2.5.1
collected 129 items                                                                                                                                                                                               

tests/test_align.py ..........................                                                                                                                                                              [ 20%]
tests/test_crd_download.py .......                                                                                                                                                                          [ 25%]
tests/test_crd_normalize.py .                                                                                                                                                                               [ 26%]
tests/test_crmprtd.py ...                                                                                                                                                                                   [ 28%]
tests/test_download.py ......                                                                                                                                                                               [ 33%]
tests/test_ec.py .............                                                                                                                                                                              [ 43%]
tests/test_ec_normalize.py ...                                                                                                                                                                              [ 45%]
tests/test_ec_swob_download.py .....                                                                                                                                                                        [ 49%]
tests/test_ec_swob_normalize.py .....                                                                                                                                                                       [ 53%]
tests/test_fixture.py ....                                                                                                                                                                                  [ 56%]
tests/test_insert.py .................                                                                                                                                                                      [ 69%]
tests/test_moti.py .....                                                                                                                                                                                    [ 73%]
tests/test_moti_download.py ............                                                                                                                                                                    [ 82%]
tests/test_moti_normalize.py ..........                                                                                                                                                                     [ 90%]
tests/test_sqlalchemy.py ...                                                                                                                                                                                [ 93%]
tests/test_wamr_normalize.py .....                                                                                                                                                                          [ 96%]
tests/test_wmb_download.py .                                                                                                                                                                                [ 97%]
tests/test_wmb_normalize.py ...                                                                                                                                                                             [100%]

=========================================================================================== 129 passed in 97.44 seconds ===========================================================================================
(venv) nrados@pcic-2024:~/github/pcic/crmprtd$ pytest -m 
usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: argument -m: expected one argument
(venv) nrados@pcic-2024:~/github/pcic/crmprtd$ pytest -m "not slow"
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.9, pytest-3.6.3, py-1.6.0, pluggy-0.6.0
rootdir: /home/nrados/github/pcic/crmprtd, inifile: pytest.ini
plugins: requests-mock-1.7.0, mock-1.11.1, flake8-1.0.1, cov-2.5.1
collected 129 items / 41 deselected                                                                                                                                                                               

tests/test_align.py ...                                                                                                                                                                                     [  3%]
tests/test_crd_download.py .......                                                                                                                                                                          [ 11%]
tests/test_crd_normalize.py .                                                                                                                                                                               [ 12%]
tests/test_crmprtd.py ...                                                                                                                                                                                   [ 15%]
tests/test_download.py ......                                                                                                                                                                               [ 22%]
tests/test_ec.py ............                                                                                                                                                                               [ 36%]
tests/test_ec_normalize.py ...                                                                                                                                                                              [ 39%]
tests/test_ec_swob_download.py .....                                                                                                                                                                        [ 45%]
tests/test_ec_swob_normalize.py .....                                                                                                                                                                       [ 51%]
tests/test_insert.py .......                                                                                                                                                                                [ 59%]
tests/test_moti.py .....                                                                                                                                                                                    [ 64%]
tests/test_moti_download.py ............                                                                                                                                                                    [ 78%]
tests/test_moti_normalize.py ..........                                                                                                                                                                     [ 89%]
tests/test_wamr_normalize.py .....                                                                                                                                                                          [ 95%]
tests/test_wmb_download.py .                                                                                                                                                                                [ 96%]
tests/test_wmb_normalize.py ...                                                                                                                                                                             [100%]

==================================================================================== 88 passed, 41 deselected in 1.20 seconds =====================================================================================
```

Resolves #71

nikola-rados commented:
@jameshiebert I actually jumped the gun on this a bit.  I'll be changing it to a draft until it is ready.

nikola-rados commented:
@jameshiebert this is ready for review now.  This is my first stop for the overall DevOps update and I'm hoping we can set a good standard for what the CI for python repos will look like.

jameshiebert commented on README.md:
````diff
@@ -92,15 +93,12 @@ py.test -v tests
-  - `python setup.py sdist bdist_wheel`, then copy the `dist/*` to the pypiserver, OR
-  - `python setup.py sdist upload -r <server>` if you have that set up in your `.pypirc` file
+```
+4. Actions will build and release the package
````
Can we specify "GH Actions", in case a dev who is not familiar with the context reads this?

jameshiebert replied:
And it would probably be good to guide the reader to the actions config file.

nikola-rados replied:
:+1:

jameshiebert commented on README.md:
````diff
@@ -92,15 +93,12 @@ py.test -v tests
-1. Summarize release changes in `NEWS.md`
-1. Commit these changes, then tag the release
-  ```bash
+2. Summarize release changes in `NEWS.md`
````
Technically, in Markdown [numbers are ignored](https://www.markdownguide.org/basic-syntax#ordered-lists), but this change is fine.

nikola-rados replied:
I had no idea!

jameshiebert commented on pytest.ini:
```diff
@@ -0,0 +1,3 @@
+[pytest]
+markers =
+    slow: marks tests that are slow (deselect with '-m "not slow"')
```
Do we have an approximate time threshold for "slow"?

nikola-rados replied:
I tried to start a conversation about it in our [team chat](https://pcic.zulipchat.com/#narrow/stream/227425-CSG-crmprtd/topic/Testing/near/202372786) but didn't get any feedback. In this instance I just chose all the tests that took longer than 1 second.  Here is a look at the timing outputs:
```bash
(venv) nrados@pcic-2024:~/github/pcic/crmprtd$ python3 -m pytest --duration=0
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.9, pytest-3.6.3, py-1.6.0, pluggy-0.6.0
rootdir: /home/nrados/github/pcic/crmprtd, inifile: pytest.ini
plugins: requests-mock-1.7.0, mock-1.11.1, flake8-1.0.1, cov-2.5.1
collected 129 items                                                                                                                                                                                               

tests/test_align.py ..........................                                                                                                                                                              [ 20%]
tests/test_crd_download.py .......                                                                                                                                                                          [ 25%]
tests/test_crd_normalize.py .                                                                                                                                                                               [ 26%]
tests/test_crmprtd.py ...                                                                                                                                                                                   [ 28%]
tests/test_download.py ......                                                                                                                                                                               [ 33%]
tests/test_ec.py .............                                                                                                                                                                              [ 43%]
tests/test_ec_normalize.py ...                                                                                                                                                                              [ 45%]
tests/test_ec_swob_download.py .....                                                                                                                                                                        [ 49%]
tests/test_ec_swob_normalize.py .....                                                                                                                                                                       [ 53%]
tests/test_fixture.py ....                                                                                                                                                                                  [ 56%]
tests/test_insert.py .................                                                                                                                                                                      [ 69%]
tests/test_moti.py .....                                                                                                                                                                                    [ 73%]
tests/test_moti_download.py ............                                                                                                                                                                    [ 82%]
tests/test_moti_normalize.py ..........                                                                                                                                                                     [ 90%]
tests/test_sqlalchemy.py ...                                                                                                                                                                                [ 93%]
tests/test_wamr_normalize.py .....                                                                                                                                                                          [ 96%]
tests/test_wmb_download.py .                                                                                                                                                                                [ 97%]
tests/test_wmb_normalize.py ...                                                                                                                                                                             [100%]

============================================================================================= slowest test durations ==============================================================================================
4.04s setup    tests/test_align.py::test_is_network[FLNRO-WMB-True]
3.63s setup    tests/test_align.py::test_is_network[WMB-False]
3.63s setup    tests/test_align.py::test_get_history_with_single_match
3.61s setup    tests/test_align.py::test_align_failures[obs_tuple3]
3.60s setup    tests/test_align.py::test_unit_check[EC_raw-precipitation-cm-10-100]
3.59s setup    tests/test_align.py::test_get_history_with_no_matches
3.58s setup    tests/test_insert.py::test_bisect_insert_strategy[duplicates-days1-1]
3.57s setup    tests/test_align.py::test_align_failures[obs_tuple5]
3.56s setup    tests/test_insert.py::test_obs_exist_in_db
3.55s setup    tests/test_align.py::test_align_successes[obs_tuple2-1-expected_time2-1-10]
3.54s setup    tests/test_align.py::test_get_variable_no_match
3.54s setup    tests/test_insert.py::test_mass_insert_obs_weird
3.54s setup    tests/test_align.py::test_align_failures[obs_tuple2]
3.52s setup    tests/test_align.py::test_align_successes[obs_tuple1-20-expected_time1-2-100]
3.52s setup    tests/test_insert.py::test_single_insert_obs
3.52s setup    tests/test_align.py::test_align_failures[obs_tuple4]
3.51s setup    tests/test_align.py::test_align_successes[obs_tuple0-21-expected_time0-2-123]
3.51s setup    tests/test_align.py::test_get_variable
3.50s setup    tests/test_align.py::test_get_history_with_multiple_matches_and_location
3.50s setup    tests/test_insert.py::test_bisect_insert_strategy[none-days2-0]
3.50s setup    tests/test_align.py::test_closest_stns_within_threshold
3.50s setup    tests/test_align.py::test_align_failures[obs_tuple1]
3.50s setup    tests/test_sqlalchemy.py::test_nested_transactions_2
3.49s setup    tests/test_insert.py::test_contains_all_duplicates_no_dup
3.48s setup    tests/test_align.py::test_align_successes[obs_tuple4-5-expected_time4-2-123]
3.48s setup    tests/test_insert.py::test_bisect_insert_strategy[unique-days0-7]
3.48s setup    tests/test_ec.py::test_OmMember_index_error_handle
3.48s setup    tests/test_insert.py::test_obs_exist_not_in_db
3.47s setup    tests/test_align.py::test_align_failures[obs_tuple0]
3.47s setup    tests/test_insert.py::test_contains_all_duplicates_all_dup
3.46s setup    tests/test_align.py::test_unit_check[FLNRO-WMB-relative_humidity-percent-10-10]
3.45s setup    tests/test_insert.py::test_single_insert_obs_not_unique
3.44s setup    tests/test_align.py::test_closest_stns_within_threshold_bad_data
3.43s setup    tests/test_align.py::test_align_successes[obs_tuple3-5-expected_time3-1-10]
3.41s setup    tests/test_align.py::test_get_history_with_multiple_matches_and_no_location
3.41s setup    tests/test_sqlalchemy.py::test_nested_transactions_3
3.40s setup    tests/test_fixture.py::test_db_has_data
3.38s setup    tests/test_sqlalchemy.py::test_nested_transactions_1
3.36s setup    tests/test_fixture.py::test_can_instantiate
3.24s setup    tests/test_fixture.py::test_db_has_geo
3.17s setup    tests/test_fixture.py::test_db_has_binary
0.17s teardown tests/test_align.py::test_align_failures[obs_tuple3]
0.17s teardown tests/test_insert.py::test_single_insert_obs
0.17s teardown tests/test_align.py::test_align_successes[obs_tuple4-5-expected_time4-2-123]
0.17s teardown tests/test_fixture.py::test_db_has_data
0.17s teardown tests/test_align.py::test_is_network[FLNRO-WMB-True]
0.17s teardown tests/test_align.py::test_align_successes[obs_tuple0-21-expected_time0-2-123]
0.17s teardown tests/test_sqlalchemy.py::test_nested_transactions_3
0.17s teardown tests/test_insert.py::test_obs_exist_not_in_db
0.17s teardown tests/test_insert.py::test_bisect_insert_strategy[none-days2-0]
0.17s teardown tests/test_align.py::test_get_history_with_single_match
0.16s teardown tests/test_align.py::test_closest_stns_within_threshold
0.16s teardown tests/test_align.py::test_align_successes[obs_tuple1-20-expected_time1-2-100]
0.16s teardown tests/test_align.py::test_unit_check[FLNRO-WMB-relative_humidity-percent-10-10]
0.16s teardown tests/test_insert.py::test_bisect_insert_strategy[unique-days0-7]
0.16s teardown tests/test_align.py::test_unit_check[EC_raw-precipitation-cm-10-100]
0.16s teardown tests/test_align.py::test_align_successes[obs_tuple2-1-expected_time2-1-10]
0.16s teardown tests/test_align.py::test_closest_stns_within_threshold_bad_data
0.16s teardown tests/test_align.py::test_is_network[WMB-False]
0.16s teardown tests/test_align.py::test_get_variable_no_match
0.16s teardown tests/test_insert.py::test_obs_exist_in_db
0.16s teardown tests/test_insert.py::test_bisect_insert_strategy[duplicates-days1-1]
0.16s teardown tests/test_align.py::test_align_failures[obs_tuple0]
0.16s teardown tests/test_align.py::test_get_variable
0.16s teardown tests/test_insert.py::test_contains_all_duplicates_no_dup
0.16s teardown tests/test_align.py::test_align_failures[obs_tuple1]
0.16s teardown tests/test_insert.py::test_single_insert_obs_not_unique
0.16s teardown tests/test_insert.py::test_mass_insert_obs_weird
0.16s teardown tests/test_ec.py::test_OmMember_index_error_handle
0.16s teardown tests/test_insert.py::test_contains_all_duplicates_all_dup
0.16s teardown tests/test_sqlalchemy.py::test_nested_transactions_2
0.16s teardown tests/test_align.py::test_get_history_with_multiple_matches_and_location
0.16s teardown tests/test_align.py::test_align_failures[obs_tuple2]
0.16s teardown tests/test_align.py::test_get_history_with_no_matches
0.16s teardown tests/test_sqlalchemy.py::test_nested_transactions_1
0.16s teardown tests/test_align.py::test_align_failures[obs_tuple4]
0.16s teardown tests/test_align.py::test_align_successes[obs_tuple3-5-expected_time3-1-10]
0.16s teardown tests/test_align.py::test_get_history_with_multiple_matches_and_no_location
0.16s teardown tests/test_align.py::test_align_failures[obs_tuple5]
0.16s teardown tests/test_fixture.py::test_can_instantiate
0.15s teardown tests/test_fixture.py::test_db_has_geo
0.15s teardown tests/test_fixture.py::test_db_has_binary
0.12s call     tests/test_insert.py::test_bisect_insert_strategy[duplicates-days1-1]
0.11s call     tests/test_align.py::test_align_failures[obs_tuple1]
0.06s call     tests/test_ec_swob_normalize.py::test_normalize[normalize0]
0.06s call     tests/test_ec_swob_download.py::test_get_url_list
0.06s call     tests/test_ec_swob_normalize.py::test_normalize[normalize1]
0.05s call     tests/test_align.py::test_align_successes[obs_tuple4-5-expected_time4-2-123]
0.04s call     tests/test_align.py::test_align_successes[obs_tuple1-20-expected_time1-2-100]
0.04s call     tests/test_align.py::test_align_successes[obs_tuple0-21-expected_time0-2-123]
0.04s call     tests/test_align.py::test_align_failures[obs_tuple0]
0.04s call     tests/test_align.py::test_get_history_with_multiple_matches_and_location
0.03s call     tests/test_align.py::test_align_failures[obs_tuple2]
0.03s call     tests/test_align.py::test_align_successes[obs_tuple3-5-expected_time3-1-10]
0.03s call     tests/test_align.py::test_get_history_with_single_match
0.03s call     tests/test_align.py::test_get_history_with_no_matches
0.03s call     tests/test_ec_swob_normalize.py::test_normalize[normalize3]
0.03s call     tests/test_ec_swob_normalize.py::test_normalize[normalize2]
0.03s call     tests/test_align.py::test_align_failures[obs_tuple5]
0.03s call     tests/test_insert.py::test_mass_insert_obs_weird
0.03s call     tests/test_align.py::test_get_history_with_multiple_matches_and_no_location
0.03s call     tests/test_insert.py::test_bisect_insert_strategy[unique-days0-7]
0.03s call     tests/test_align.py::test_align_successes[obs_tuple2-1-expected_time2-1-10]
0.03s call     tests/test_insert.py::test_contains_all_duplicates_all_dup
0.02s call     tests/test_wmb_download.py::test_setup_logging
0.02s call     tests/test_align.py::test_closest_stns_within_threshold_bad_data
0.02s call     tests/test_insert.py::test_single_insert_obs
0.01s call     tests/test_ec_normalize.py::test_normalize_good_data
0.01s call     tests/test_align.py::test_unit_check[EC_raw-precipitation-cm-10-100]
0.01s call     tests/test_insert.py::test_bisect_insert_strategy[none-days2-0]
0.01s call     tests/test_insert.py::test_contains_all_duplicates_no_dup
0.01s call     tests/test_ec_normalize.py::test_normalize_bad_date
0.01s call     tests/test_insert.py::test_single_insert_obs_not_unique
0.01s call     tests/test_align.py::test_align_failures[obs_tuple3]
0.01s call     tests/test_ec_normalize.py::test_normalize_no_station_id
0.01s call     tests/test_sqlalchemy.py::test_nested_transactions_3
0.01s call     tests/test_crd_normalize.py::test_normalize_good_data
0.01s call     tests/test_align.py::test_closest_stns_within_threshold
0.01s call     tests/test_insert.py::test_obs_exist_not_in_db
0.01s call     tests/test_align.py::test_is_network[WMB-False]
0.01s call     tests/test_sqlalchemy.py::test_nested_transactions_2
0.01s call     tests/test_download.py::test_https_download
0.01s call     tests/test_insert.py::test_obs_exist_in_db
0.01s call     tests/test_align.py::test_get_variable
0.01s call     tests/test_align.py::test_is_network[FLNRO-WMB-True]
0.01s call     tests/test_align.py::test_get_variable_no_match
0.01s call     tests/test_align.py::test_unit_check[FLNRO-WMB-relative_humidity-percent-10-10]
0.00s call     tests/test_sqlalchemy.py::test_nested_transactions_1
0.00s call     tests/test_wmb_normalize.py::test_normalize_good_data
0.00s call     tests/test_ec.py::test_xsl_transform_tendency_amount[x0--0.12]
0.00s call     tests/test_ec_swob_normalize.py::test_normalize_missing_values
0.00s call     tests/test_wamr_normalize.py::test_normalize_good_data
0.00s call     tests/test_fixture.py::test_db_has_data
0.00s call     tests/test_moti.py::test_var_transforms_all
0.00s call     tests/test_moti_download.py::test_verify_date
0.00s call     tests/test_moti_download.py::test_download[None-etime2-the_koots-expected_payload2]
0.00s call     tests/test_moti.py::test_var_transforms
0.00s call     tests/test_moti_download.py::test_download[None-None-the_koots-expected_payload1]
0.00s call     tests/test_moti_download.py::test_download[2020/06/30 11:53:11-2020/06/30 11:53:11-the_koots-expected_payload4]
0.00s call     tests/test_wmb_normalize.py::test_normalize_bad_value
0.00s call     tests/test_download.py::test_https_download_404
0.00s call     tests/test_moti_normalize.py::test_normalize_good_data
0.00s call     tests/test_moti_download.py::test_download[stime3-None-the_koots-expected_payload3]
0.00s call     tests/test_align.py::test_convert_unit[Deg.-degree]
0.00s call     tests/test_moti_normalize.py::test_normalize_empty_data[<observation-series></observation-series>]
0.00s call     tests/test_moti_download.py::test_download[None-None-None-expected_payload0]
0.00s call     tests/test_ec.py::test_OmMember_index_error_handle
0.00s call     tests/test_crd_download.py::test_verify_dates[2020/06/06-2020/06/06]
0.00s setup    tests/test_crmprtd.py::test_subset_dict[a_dict2-keys2-expected2]
0.00s setup    tests/test_crd_download.py::test_verify_dates[2020/01/01-2019/12/31]
0.00s call     tests/test_ec.py::test_xsl_transform_tendency_amount[x1-0.12]
0.00s call     tests/test_align.py::test_convert_unit[\xb0C-celsius]
0.00s call     tests/test_moti_normalize.py::test_normalize_missing_var_name
0.00s call     tests/test_ec.py::test_xsl_transform_cloud_cover[x0-8]
0.00s setup    tests/test_moti_download.py::test_download[None-None-None-expected_payload0]
0.00s setup    tests/test_ec_swob_download.py::test_get_url_list
0.00s call     tests/test_ec.py::test_xsl_transform_wind_direction[x0-270]
0.00s setup    tests/test_moti_download.py::test_download[stime3-None-the_koots-expected_payload3]
0.00s call     tests/test_ec.py::test_xsl_transform_wind_direction[x1-135]
0.00s call     tests/test_moti_normalize.py::test_normalize_bad_value
0.00s call     tests/test_moti_normalize.py::test_normalize_missing_time
0.00s call     tests/test_fixture.py::test_db_has_geo
0.00s call     tests/test_download.py::test_extract_auth[None-None-expected3]
0.00s call     tests/test_moti_normalize.py::test_normalize_missing_value
0.00s setup    tests/test_insert.py::test_chunks[1200-expected0]
0.00s setup    tests/test_moti_download.py::test_download[None-None-the_koots-expected_payload1]
0.00s setup    tests/test_moti_download.py::test_download[None-etime2-the_koots-expected_payload2]
0.00s setup    tests/test_ec_swob_download.py::test_match_date[/20190101/station_id/-False]
0.00s setup    tests/test_crd_download.py::test_verify_dates[None-2011/11/11]
0.00s setup    tests/test_moti_download.py::test_download[2020/06/30 11:53:11-2020/06/30 11:53:11-the_koots-expected_payload4]
0.00s call     tests/test_fixture.py::test_db_has_binary
0.00s call     tests/test_crd_download.py::test_verify_dates[2010/01/01-2010/01/29 01:00:00]
0.00s call     tests/test_moti_download.py::test_verify_date_exception[not-a-datestring]
0.00s teardown tests/test_moti_download.py::test_download[None-etime2-the_koots-expected_payload2]
0.00s teardown tests/test_crmprtd.py::test_subset_dict[a_dict2-keys2-expected2]
0.00s setup    tests/test_crmprtd.py::test_subset_dict[a_dict0-keys0-expected0]
0.00s setup    tests/test_crd_download.py::test_verify_dates[2011/11/11-None]
0.00s call     tests/test_wamr_normalize.py::test_normalize_unpack_error_catch
0.00s call     tests/test_moti_normalize.py::test_normalize_bad_time
0.00s setup    tests/test_crd_download.py::test_make_url[None-None-https://webservices.crd.bc.ca/weatherdata/PCIC/]
0.00s setup    tests/test_moti.py::test_var_transforms
0.00s teardown tests/test_wmb_normalize.py::test_normalize_bad_value
0.00s setup    tests/test_crd_download.py::test_make_url[sdate1-edate1-https://webservices.crd.bc.ca/weatherdata/PCIC/201906060000-201906070000]
0.00s call     tests/test_crd_download.py::test_verify_dates[2020/01/01-2019/12/31]
0.00s teardown tests/test_crd_download.py::test_verify_dates[2011/11/11-None]
0.00s setup    tests/test_insert.py::test_split[tuple0-expected_a0-expected_b0]
0.00s setup    tests/test_align.py::test_convert_unit[Deg.-degree]
0.00s setup    tests/test_crmprtd.py::test_subset_dict[a_dict1-keys1-expected1]
0.00s call     tests/test_moti.py::test_url_generator
0.00s setup    tests/test_moti.py::test_var_transforms_all
0.00s setup    tests/test_crd_download.py::test_verify_dates[2010/01/01-2010/01/29 01:00:00]
0.00s setup    tests/test_crd_download.py::test_verify_dates[2020/06/06-2020/06/06]
0.00s call     tests/test_moti_download.py::test_verify_date_exception[None]
0.00s setup    tests/test_insert.py::test_split[tuple1-1-expected_b1]
0.00s call     tests/test_moti_normalize.py::test_normalize_missing_stn_indexerror
0.00s call     tests/test_crd_download.py::test_verify_dates[2011/11/11-None]
0.00s setup    tests/test_download.py::test_extract_auth[foo-bar-expected0]
0.00s setup    tests/test_ec_swob_download.py::test_match_date[/20191018/station_id/-True]
0.00s call     tests/test_align.py::test_convert_unit[Deg-degree]
0.00s setup    tests/test_insert.py::test_get_sample_indices[10-500-10]
0.00s setup    tests/test_align.py::test_convert_unit[Deg-degree]
0.00s setup    tests/test_insert.py::test_get_sample_indices[500-10-10]
0.00s setup    tests/test_insert.py::test_split[tuple2-expected_a2-expected_b2]
0.00s teardown tests/test_ec_swob_download.py::test_match_date[/20190101/station_id/-False]
0.00s call     tests/test_wamr_normalize.py::test_normalize_bad_date
0.00s setup    tests/test_align.py::test_convert_unit[\xb0C-celsius]
0.00s call     tests/test_ec_swob_download.py::test_match_date[/20191018/station_id/-True]
0.00s setup    tests/test_ec_swob_download.py::test_match_date[/url_without_any_date/-True]
0.00s call     tests/test_crd_download.py::test_verify_dates[None-2011/11/11]
0.00s setup    tests/test_download.py::test_extract_auth[foo-None-expected1]
0.00s setup    tests/test_download.py::test_extract_auth[None-bar-expected2]
0.00s teardown tests/test_crd_download.py::test_verify_dates[2020/06/06-2020/06/06]
0.00s teardown tests/test_moti_download.py::test_download[stime3-None-the_koots-expected_payload3]
0.00s setup    tests/test_download.py::test_https_download
0.00s call     tests/test_moti_normalize.py::test_normalize_empty_data[<observation-series>\n     </observation-series>]
0.00s call     tests/test_insert.py::test_get_sample_indices[500-10-10]
0.00s teardown tests/test_moti_normalize.py::test_normalize_empty_data[<observation-series></observation-series>]
0.00s call     tests/test_moti_normalize.py::test_normalize_empty_data[<observation-series />]
0.00s setup    tests/test_ec.py::test_makeurl_no_time_hourly
0.00s call     tests/test_ec.py::test_makeurl_no_time_hourly
0.00s setup    tests/test_insert.py::test_chunks[1201-expected1]
0.00s teardown tests/test_ec_swob_normalize.py::test_normalize[normalize0]
0.00s teardown tests/test_crd_download.py::test_verify_dates[2010/01/01-2010/01/29 01:00:00]
0.00s setup    tests/test_moti_download.py::test_verify_date_exception[not-a-datestring]
0.00s setup    tests/test_wamr_normalize.py::test_normalize_bad_date
0.00s teardown tests/test_align.py::test_convert_unit[Deg.-degree]
0.00s call     tests/test_ec_swob_download.py::test_match_date[/20190101/station_id/-False]
0.00s setup    tests/test_ec.py::test_makeurl[daily-BC-EN-args0-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/yesterday/yesterday_bc_20160115_e.xml]
0.00s call     tests/test_insert.py::test_chunks[1201-expected1]
0.00s teardown tests/test_crmprtd.py::test_subset_dict[a_dict0-keys0-expected0]
0.00s teardown tests/test_ec_normalize.py::test_normalize_bad_date
0.00s call     tests/test_insert.py::test_chunks[1200-expected0]
0.00s setup    tests/test_ec_swob_normalize.py::test_normalize[normalize1]
0.00s setup    tests/test_moti_download.py::test_verify_date_exception[None]
0.00s call     tests/test_moti.py::test_url_generator_truncates
0.00s teardown tests/test_align.py::test_convert_unit[\xb0C-celsius]
0.00s teardown tests/test_moti_download.py::test_download[None-None-None-expected_payload0]
0.00s teardown tests/test_ec.py::test_makeurl_no_time_hourly
0.00s teardown tests/test_moti_download.py::test_download[None-None-the_koots-expected_payload1]
0.00s teardown tests/test_crd_download.py::test_verify_dates[2020/01/01-2019/12/31]
0.00s setup    tests/test_wamr_normalize.py::test_normalize_good_data
0.00s teardown tests/test_crd_download.py::test_make_url[sdate1-edate1-https://webservices.crd.bc.ca/weatherdata/PCIC/201906060000-201906070000]
0.00s teardown tests/test_crd_download.py::test_make_url[None-None-https://webservices.crd.bc.ca/weatherdata/PCIC/]
0.00s setup    tests/test_moti_download.py::test_download_bad_args[None-etime0]
0.00s setup    tests/test_download.py::test_extract_auth[None-None-expected3]
0.00s teardown tests/test_crd_download.py::test_verify_dates[None-2011/11/11]
0.00s teardown tests/test_moti.py::test_var_transforms_all
0.00s teardown tests/test_crmprtd.py::test_subset_dict[a_dict1-keys1-expected1]
0.00s teardown tests/test_align.py::test_convert_unit[Deg-degree]
0.00s call     tests/test_align.py::test_align_failures[obs_tuple4]
0.00s call     tests/test_moti.py::test_url_generator_backwards_dates
0.00s setup    tests/test_moti_download.py::test_download_bad_args[stime1-None]
0.00s setup    tests/test_moti_download.py::test_download_bad_args[stime2-etime2]
0.00s setup    tests/test_ec.py::test_xsl_transform_tendency_amount[x0--0.12]
0.00s setup    tests/test_ec.py::test_makeurl[hourly-BC-EN-args1-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/hourly/hourly_bc_2016011521_e.xml]
0.00s call     tests/test_wamr_normalize.py::test_normalize_bad_value
0.00s setup    tests/test_ec.py::test_makeurl[nofreq-BC-EN-args2-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/yesterday/yesterday_bc_20160115_e.xml]
0.00s setup    tests/test_crd_normalize.py::test_normalize_good_data
0.00s setup    tests/test_moti.py::test_url_generator
0.00s teardown tests/test_crd_normalize.py::test_normalize_good_data
0.00s setup    tests/test_ec.py::test_makeurl[hourly-BC-nolang-args4-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/hourly/hourly_bc_2016011521_e.xml]
0.00s setup    tests/test_ec.py::test_makeurl[hourly-noprov-EN-args3-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/hourly/hourly_bc_2016011521_e.xml]
0.00s setup    tests/test_ec_normalize.py::test_normalize_good_data
0.00s setup    tests/test_moti_normalize.py::test_normalize_empty_data[<observation-series></observation-series>]
0.00s setup    tests/test_moti_normalize.py::test_normalize_empty_data[<observation-series />]
0.00s teardown tests/test_download.py::test_extract_auth[foo-bar-expected0]
0.00s setup    tests/test_ec.py::test_xsl_transform_cloud_cover[x0-8]
0.00s teardown tests/test_moti_download.py::test_download[2020/06/30 11:53:11-2020/06/30 11:53:11-the_koots-expected_payload4]
0.00s setup    tests/test_wmb_download.py::test_setup_logging
0.00s call     tests/test_crd_download.py::test_make_url[sdate1-edate1-https://webservices.crd.bc.ca/weatherdata/PCIC/201906060000-201906070000]
0.00s setup    tests/test_moti_download.py::test_verify_date
0.00s teardown tests/test_ec_swob_download.py::test_match_date[/20191018/station_id/-True]
0.00s teardown tests/test_moti.py::test_url_generator_backwards_dates
0.00s teardown tests/test_ec_swob_download.py::test_match_date[/url_without_any_date/-True]
0.00s setup    tests/test_ec.py::test_xsl_transform_tendency_amount[x1-0.12]
0.00s setup    tests/test_download.py::test_https_download_404
0.00s setup    tests/test_moti_normalize.py::test_normalize_empty_data[<observation-series>\n     </observation-series>]
0.00s setup    tests/test_moti_normalize.py::test_normalize_bad_time
0.00s teardown tests/test_insert.py::test_split[tuple0-expected_a0-expected_b0]
0.00s teardown tests/test_insert.py::test_split[tuple1-1-expected_b1]
0.00s call     tests/test_wmb_normalize.py::test_normalize_bad_date
0.00s teardown tests/test_insert.py::test_get_sample_indices[500-10-10]
0.00s setup    tests/test_ec.py::test_xsl_transform_wind_direction[x1-135]
0.00s setup    tests/test_wmb_normalize.py::test_normalize_good_data
0.00s setup    tests/test_ec.py::test_xsl_transform_wind_direction[x0-270]
0.00s teardown tests/test_insert.py::test_get_sample_indices[10-500-10]
0.00s teardown tests/test_insert.py::test_split[tuple2-expected_a2-expected_b2]
0.00s teardown tests/test_download.py::test_https_download
0.00s call     tests/test_crmprtd.py::test_subset_dict[a_dict2-keys2-expected2]
0.00s teardown tests/test_insert.py::test_chunks[1201-expected1]
0.00s teardown tests/test_ec_normalize.py::test_normalize_good_data
0.00s teardown tests/test_ec_normalize.py::test_normalize_no_station_id
0.00s setup    tests/test_wamr_normalize.py::test_normalize_unpack_error_catch
0.00s call     tests/test_ec_swob_download.py::test_match_date[/url_without_any_date/-True]
0.00s setup    tests/test_wamr_normalize.py::test_normalize_missing_value
0.00s teardown tests/test_insert.py::test_chunks[1200-expected0]
0.00s setup    tests/test_ec_normalize.py::test_normalize_no_station_id
0.00s call     tests/test_crd_download.py::test_make_url[None-None-https://webservices.crd.bc.ca/weatherdata/PCIC/]
0.00s call     tests/test_crmprtd.py::test_subset_dict[a_dict0-keys0-expected0]
0.00s call     tests/test_ec_swob_download.py::test_split_multi_xml_stream
0.00s teardown tests/test_moti_download.py::test_verify_date_exception[not-a-datestring]
0.00s teardown tests/test_ec.py::test_xsl_transform_tendency_amount[x0--0.12]
0.00s teardown tests/test_wamr_normalize.py::test_normalize_bad_date
0.00s setup    tests/test_ec_normalize.py::test_normalize_bad_date
0.00s teardown tests/test_moti_download.py::test_verify_date_exception[None]
0.00s setup    tests/test_ec_swob_normalize.py::test_normalize[normalize0]
0.00s call     tests/test_crmprtd.py::test_subset_dict[a_dict1-keys1-expected1]
0.00s teardown tests/test_wamr_normalize.py::test_normalize_good_data
0.00s setup    tests/test_ec_swob_normalize.py::test_normalize_missing_values
0.00s teardown tests/test_wmb_download.py::test_setup_logging
0.00s call     tests/test_insert.py::test_get_sample_indices[10-500-10]
0.00s setup    tests/test_wamr_normalize.py::test_normalize_bad_value
0.00s call     tests/test_moti_download.py::test_download_too_long
0.00s setup    tests/test_ec_swob_normalize.py::test_normalize[normalize3]
0.00s call     tests/test_wamr_normalize.py::test_normalize_missing_value
0.00s setup    tests/test_moti.py::test_url_generator_truncates
0.00s teardown tests/test_download.py::test_extract_auth[foo-None-expected1]
0.00s teardown tests/test_wamr_normalize.py::test_normalize_unpack_error_catch
0.00s setup    tests/test_wmb_normalize.py::test_normalize_bad_date
0.00s setup    tests/test_moti.py::test_url_generator_backwards_dates
0.00s setup    tests/test_wmb_normalize.py::test_normalize_bad_value
0.00s setup    tests/test_ec_swob_normalize.py::test_normalize[normalize2]
0.00s call     tests/test_moti_download.py::test_download_bad_args[None-etime0]
0.00s teardown tests/test_moti_download.py::test_verify_date
0.00s teardown tests/test_wamr_normalize.py::test_normalize_missing_value
0.00s call     tests/test_insert.py::test_split[tuple0-expected_a0-expected_b0]
0.00s teardown tests/test_ec.py::test_makeurl[hourly-BC-nolang-args4-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/hourly/hourly_bc_2016011521_e.xml]
0.00s teardown tests/test_moti.py::test_var_transforms
0.00s call     tests/test_fixture.py::test_can_instantiate
0.00s teardown tests/test_download.py::test_extract_auth[None-bar-expected2]
0.00s teardown tests/test_ec_swob_normalize.py::test_normalize[normalize2]
0.00s teardown tests/test_wmb_normalize.py::test_normalize_good_data
0.00s call     tests/test_moti_download.py::test_download_bad_args[stime1-None]
0.00s teardown tests/test_ec_swob_download.py::test_get_url_list
0.00s call     tests/test_download.py::test_extract_auth[foo-bar-expected0]
0.00s call     tests/test_insert.py::test_split[tuple2-expected_a2-expected_b2]
0.00s teardown tests/test_moti.py::test_url_generator
0.00s teardown tests/test_ec_swob_normalize.py::test_normalize[normalize3]
0.00s call     tests/test_insert.py::test_split[tuple1-1-expected_b1]
0.00s teardown tests/test_moti.py::test_url_generator_truncates
0.00s teardown tests/test_wamr_normalize.py::test_normalize_bad_value
0.00s teardown tests/test_moti_download.py::test_download_bad_args[stime1-None]
0.00s setup    tests/test_ec.py::test_makeurl_no_time_daily
0.00s teardown tests/test_wmb_normalize.py::test_normalize_bad_date
0.00s teardown tests/test_ec.py::test_xsl_transform_wind_direction[x0-270]
0.00s teardown tests/test_download.py::test_extract_auth[None-None-expected3]
0.00s teardown tests/test_ec.py::test_xsl_transform_cloud_cover[x0-8]
0.00s call     tests/test_moti_download.py::test_download_bad_args[stime2-etime2]
0.00s teardown tests/test_moti_download.py::test_download_bad_args[None-etime0]
0.00s teardown tests/test_ec.py::test_xsl_transform_tendency_amount[x1-0.12]
0.00s teardown tests/test_ec.py::test_makeurl[daily-BC-EN-args0-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/yesterday/yesterday_bc_20160115_e.xml]
0.00s teardown tests/test_ec.py::test_xsl_transform_wind_direction[x1-135]
0.00s call     tests/test_download.py::test_extract_auth[foo-None-expected1]
0.00s teardown tests/test_moti_normalize.py::test_normalize_empty_data[<observation-series>\n     </observation-series>]
0.00s teardown tests/test_ec.py::test_makeurl[hourly-noprov-EN-args3-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/hourly/hourly_bc_2016011521_e.xml]
0.00s teardown tests/test_ec.py::test_makeurl[nofreq-BC-EN-args2-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/yesterday/yesterday_bc_20160115_e.xml]
0.00s teardown tests/test_download.py::test_https_download_404
0.00s teardown tests/test_ec.py::test_makeurl[hourly-BC-EN-args1-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/hourly/hourly_bc_2016011521_e.xml]
0.00s teardown tests/test_ec_swob_normalize.py::test_normalize[normalize1]
0.00s teardown tests/test_moti_normalize.py::test_normalize_empty_data[<observation-series />]
0.00s call     tests/test_ec.py::test_makeurl_no_time_daily
0.00s teardown tests/test_ec.py::test_makeurl_no_time_daily
0.00s setup    tests/test_moti_normalize.py::test_normalize_good_data
0.00s teardown tests/test_ec_swob_normalize.py::test_normalize_missing_values
0.00s teardown tests/test_moti_download.py::test_download_bad_args[stime2-etime2]
0.00s setup    tests/test_moti_normalize.py::test_normalize_missing_stn_indexerror
0.00s setup    tests/test_ec_swob_download.py::test_split_multi_xml_stream
0.00s setup    tests/test_moti_normalize.py::test_normalize_missing_time
0.00s call     tests/test_ec.py::test_makeurl[daily-BC-EN-args0-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/yesterday/yesterday_bc_20160115_e.xml]
0.00s teardown tests/test_moti_normalize.py::test_normalize_missing_stn_indexerror
0.00s setup    tests/test_moti_normalize.py::test_normalize_bad_value
0.00s setup    tests/test_moti_normalize.py::test_normalize_missing_var_name
0.00s teardown tests/test_moti_normalize.py::test_normalize_bad_value
0.00s setup    tests/test_moti_normalize.py::test_normalize_missing_value
0.00s call     tests/test_ec.py::test_makeurl[nofreq-BC-EN-args2-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/yesterday/yesterday_bc_20160115_e.xml]
0.00s call     tests/test_ec.py::test_makeurl[hourly-noprov-EN-args3-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/hourly/hourly_bc_2016011521_e.xml]
0.00s call     tests/test_ec.py::test_makeurl[hourly-BC-EN-args1-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/hourly/hourly_bc_2016011521_e.xml]
0.00s call     tests/test_ec.py::test_makeurl[hourly-BC-nolang-args4-http://dd.weatheroffice.ec.gc.ca/observations/xml/BC/hourly/hourly_bc_2016011521_e.xml]
0.00s teardown tests/test_moti_normalize.py::test_normalize_missing_var_name
0.00s teardown tests/test_moti_normalize.py::test_normalize_bad_time
0.00s teardown tests/test_moti_download.py::test_download_too_long
0.00s teardown tests/test_moti_normalize.py::test_normalize_good_data
0.00s setup    tests/test_moti_download.py::test_download_too_long
0.00s teardown tests/test_moti_normalize.py::test_normalize_missing_value
0.00s call     tests/test_download.py::test_extract_auth[None-bar-expected2]
0.00s teardown tests/test_moti_normalize.py::test_normalize_missing_time
0.00s teardown tests/test_ec_swob_download.py::test_split_multi_xml_stream
========================================================================================== 129 passed in 152.35 seconds ===========================================================================================
```

nikola-rados commented:
@jameshiebert I have addressed the suggestions you made.  After running the test suite a few more times I found the times to be on average under 3 seconds for every test. So there are no tests marked with "slow" at the moment.  I have left the infrastructure in place for us to mark tests in the future.

jameshiebert commented:
Works for me!
